x-pack/legacy/plugins/triggers_actions_ui/np_ready/public/application/app_dependencies.tsx

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { createContext, useContext, ReactNode } from 'react';
+import { HashRouter } from 'react-router-dom';
+import { AppDependencies } from '../../../public/shim';
+
+let DependenciesContext: React.Context<AppDependencies>;
```

Comment:
it would be nice to discuss your thoughts about how to implement this dependencies with avoiding the application state level. My goal here was to pass deps down through the components tree and application state was first that comes to my mind

Reply:
We can definitely jump on a call to discuss this 😄 
Broadly I see a big difference between module level state and application state, and I'm happy to discuss that. 👍

Reply:
Looking at the React docs on Context, it *does* look like their examples use a module level variable to hold the context (I don't like it, but can live with it) but it isn't a `let` which can be changed, rather it's a `const` whose value is imported from a very specific module - meaning you can test it better and the React component's can't change it as they please.

I think if do `createContext` in one single place (instead of the functional component which is called for every instance and every prop/state change), then it'll already be safer.

I can expand on this on a call if you'd like.